### PR TITLE
Remove OCP\App and OCP\BackgroundJob

### PR DESCRIPTION
**Pull request by CarlSchwan**
Both deprecated since NC 14

IAppManager is the replacement for OCP\App unfortunately it can't be
dependency injected in classes used by the installer otherwise the
the database connection is initialized too early

## Test plan

1. Tests work locally
2. psalm and cs:fix are happy
3. The webui seems to still work :)

### CODE REVIEW

**Review by come-nc — APPROVED** [review]
BackgroundJob was unused?
Any chance of removing OC_App?

**Comment by CarlSchwan** [comment]
> BackgroundJob was unused?

`OCP\BackgroundJob` was unused and did nothing. This is not the same thing as `OCP\BackgroundJob\IJob` and co

> Any chance of removing OC_App?

I wish but let's do that step by step ;) there is a few api still missing in OCP\App\IAppManager to be a full replacement

File: tests/lib/Encryption/UtilTest.php [hunk 1]
```diff
@@ -4,56 +4,50 @@
...
-			->getMock();
+		$this->userManager = $this->createMock(IUserManager::class);
+		$this->groupManager = $this->createMock(IGroupManager::class);
+		$this->config = $this->createMock(IConfig::class);
 
 		$this->util = new Util(
 			$this->view,
 			$this->userManager,
 			$this->groupManager,
+			Server::get(IAppManager::class),
```
Comment (kesselb) [inline tests/lib/Encryption/UtilTest.php:50]: computer says no ;)

```
1) Test\Encryption\UtilTest::testGetEncryptionModuleId with data set #0 ('', array())
--

TypeError: 
Argument 4 passed to OC\Encryption\Util::__construct() must implement 
interface OCP\IConfig, instance of OC\App\AppManager given, called in 
/drone/src/tests/lib/Encryption/UtilTest.php on line 51
```

**Review by kesselb — CHANGES_REQUESTED** [review]

File: tests/lib/Encryption/UtilTest.php (hunk 1 above)
Comment (CarlSchwan) [reply tests/lib/Encryption/UtilTest.php]: fixed

File: core/templates/layout.base.php [hunk 2]
```diff
@@ -7,9 +7,9 @@
 		</title>
 		<meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0">
 		<meta name="theme-color" content="<?php p($theme->getColorPrimary()); ?>">
-		<link rel="icon" href="<?php print_unescaped(image_path('', 'favicon.ico')); /* IE11+ supports png */ ?>">
-		<link rel="apple-touch-icon" href="<?php print_unescaped(image_path('', 'favicon-touch.png')); ?>">
-		<link rel="mask-icon" sizes="any" href="<?php print_unescaped(image_path('', 'favicon-mask.svg')); ?>" color="<?php p($theme->getColorPrimary()); ?>">
+		<link rel="icon" href="<?php print_unescaped(image_path('core', 'favicon.ico')); /* IE11+ supports png */ ?>">
+		<link rel="apple-touch-icon" href="<?php print_unescaped(image_path('core', 'favicon-touch.png')); ?>">
+		<link rel="mask-icon" sizes="any" href="<?php print_unescaped(image_path('core', 'favicon-mask.svg')); ?>" color="<?php p($theme->getColorPrimary()); ?>">
```
Comment (kesselb) [inline core/templates/layout.base.php:12]:
```
{
  "Exception": "RuntimeException",
  "Message": "image not found: image:favicon-mask.svg webroot: serverroot:\/drone\/src",
  "Code": 0,
  "Trace": [
    {
      "file": "\/drone\/src\/lib\/private\/legacy\/template\/functions.php",
      "line": 242,
      "function": "imagePath",
      "class": "OC\\URLGenerator",
      "type": "->"
    },
    {
      "file": "\/drone\/src\/core\/templates\/layout.user.php",
      "line": 37,
      "function": "image_path"
    },
    {
      "file": "\/drone\/src\/lib\/private\/Template\/Base.php",
      "line": 180,
      "args": [
        "\/drone\/src\/core\/templates\/layout.user.php"
      ],
      "function": "include"
    },
    {
      "file": "\/drone\/src\/lib\/private\/Template\/Base.php",
      "line": 150,
      "function": "load",
      "class": "OC\\Template\\Base",
      "type": "->"
    },
    {
      "file": "\/drone\/src\/lib\/private\/legacy\/OC_Template.php",
      "line": 181,
      "function": "fetchPage",
      "class": "OC\\Template\\Base",
      "type": "->"
    },
    {
      "file": "\/drone\/src\/lib\/private\/legacy\/OC_Template.php",
      "line": 212,
      "function": "fetchPage",
      "class": "OC_Template",
      "type": "->"
    },
    {
      "file": "\/drone\/src\/lib\/public\/AppFramework\/Http\/TemplateResponse.php",
      "line": 213,
      "function": "fetchPage",
      "class": "OC_Template",
      "type": "->"
    },
    {
      "file": "\/drone\/src\/lib\/private\/AppFramework\/Http\/Dispatcher.php",
      "line": 178,
      "function": "render",
      "class": "OCP\\AppFramework\\Http\\TemplateResponse",
      "type": "->"
    },
    {
      "file": "\/drone\/src\/lib\/private\/AppFramework\/App.php",
      "line": 172,
      "function": "dispatch",
      "class": "OC\\AppFramework\\Http\\Dispatcher",
      "type": "->"
    },
    {
      "file": "\/drone\/src\/lib\/private\/Route\/Router.php",
      "line": 298,
      "function": "main",
      "class": "OC\\AppFramework\\App",
      "type": "::"
    },
    {
      "file": "\/drone\/src\/lib\/base.php",
      "line": 1021,
      "function": "match",
      "class": "OC\\Route\\Router",
      "type": "->"
    },
    {
      "file": "\/drone\/src\/index.php",
      "line": 36,
      "function": "handleRequest",
      "class": "OC",
      "type": "::"
    }
  ],
  "File": "\/drone\/src\/lib\/private\/URLGenerator.php",
  "Line": 273,
  "CustomMessage": "--"
}
```

computer is still a bit unhappy. some of the integration tests are failing.

**Review by kesselb — DISMISSED** [review]

**Review by PVince81 — APPROVED** [review]
:+1: long overdue

I suggest you grep the release tarball in case you find more usages in apps

**Comment by PVince81** [comment]
and there are conflicts

**Comment by blizzz** [comment]
CI is unhappy

**Pull request merged**